### PR TITLE
Fixes for crossSell and movingFlow for quoteCart

### PR DESCRIPTION
**Pull request by tobiasbexelius**
# Jira Issue: []

## What?
- Use `initiatedFrom` from the payload
- Quote enrichment manipulates each quote instead of the whole payload
- Merge the relevant fields from the member's data and contracts correctly

## Why?
-

## Optional checklist
- [ ] Unit tests written
- [x] Tested locally
- [x] Codescouted

### CODE REVIEW

File: src/resolvers/quote-cart/quoteCart.ts [hunk 1]
```diff
@@ -479,39 +477,24 @@ async function updateSubscribers(pubsub: RedisPubSub, id: string) {
...
-    {
+  const isYouth = activeHomeContentAgreement.typeOfContract.includes('YOUTH')
+  const isStudent =
+    activeHomeContentAgreement.typeOfContract.includes('STUDENT')
+  return {
+    ...payload,
+    ...member,
+    data: {
+      ...payload.data,
       ...member,
```
Comment (cpiehl1) [inline src/resolvers/quote-cart/quoteCart.ts:495]: Why are they being spread inside and outside of `data`? 🤔

File: src/resolvers/quote-cart/quoteCart.ts [hunk 2]
```diff
@@ -479,39 +477,23 @@ async function updateSubscribers(pubsub: RedisPubSub, id: string) {
...
-      initiatedFrom: QuoteInitiatedFromDto.CrossSell,
+  const isYouth = activeHomeContentAgreement.typeOfContract.includes('YOUTH')
+  const isStudent =
+    activeHomeContentAgreement.typeOfContract.includes('STUDENT')
+  return {
+    ...payload,
+    ...member,
+    data: {
+      ...payload.data,
+      isYouth,
```
Comment (cpiehl1) [inline src/resolvers/quote-cart/quoteCart.ts:495]: In cross-sell, we sometimes use data from the existing home content contract, such as living space. Will this follow through here? The payload is just the input to the query, right?

Comment (tobiasbexelius) [reply src/resolvers/quote-cart/quoteCart.ts:498]: Yep, fixed and tested for SE accident

**Review by cpiehl1 — APPROVED** [review]
Nice! Love the merging of cross-sell and moving-flow logic 💯

**Pull request merged**